### PR TITLE
Fix tox configuration in CI

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

- Replace coverage to pytest-cov: now We can get the proper code-coverage in CI.
image

- Remove requirement folder -> Replace to pyproject.toml
- Reduce dev optional-dependencies
- Move `create_command` from pip into install function

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,34 +17,39 @@ passenv =
...
 
 
 [testenv:unit-test]
 deps =
-    -r{toxinidir}/requirements/dev.txt
-commands =
+    .[dev]
+commands_pre =
+    ; [TODO]: Needs to be fixed so that this is not duplicated for each test run
     otx install -v
```
Comment (harimkang) [inline tox.ini:31]: @vinnamkim I still haven't found a way for tox to not always execute this command.
I'll do some more digging and try to fix it.

Comment (vinnamkim) [reply tox.ini:31]: At that time, you should remove `tox.ini` and move those configs to `pyproject.toml` as well.

**Review by vinnamkim — APPROVED** [review]
LGTM.

File: tox.ini (hunk 1 above)
Comment (harimkang) [reply tox.ini:31]: Yeah, let's take a little more time to look at it together.

**Review by jaegukhyun — APPROVED** [review]

**Review by kprokofi — APPROVED** [review]

**Pull request merged**